readme.md
# BackUpWordPress #
**Contributors:** humanmade, willmot, pauldewouters, joehoyle, mattheu, tcrsavage, cuvelier  
**Tags:** back up, backup, backups, database, zip, db, files, archive, wp-cli, humanmade  
**Requires at least:** 3.9  
**Tested up to:** 4.2-alpha  
**Stable tag:** 3.1.4  

Simple automated backups of your WordPress powered website.

## Description ##

[BackUpWordPress](https://bwp.hmn.md/?utm_source=wordpress-org&utm_medium=plugin-page&utm_campaign=freeplugin) will back up your entire site including your database and all your files on a schedule that suits you. Try it now to see how easy it is!

This plugin requires PHP version 5.3.2 or later

### Features ###

* Super simple to use, no setup required.
* Works in low memory, "shared host" environments.
* Manage multiple schedules.
* Option to have each backup file emailed to you.
* Uses `zip` and `mysqldump` for faster backups if they are available.
* Works on Linux & Windows Server.
* Exclude files and folders from your backups.
* Good support should you need help.
* Translations for Spanish, German, Chinese, Romanian, Russian, Serbian, Lithuanian, Italian, Czech, Dutch, French, Basque.

### Help develop this plugin ###

The BackUpWordPress plugin is hosted on GitHub, if you want to help out with development or testing then head over to https://github.com/humanmade/backupwordpress/.

### Translations ###

We'd also love help translating the plugin into more languages, if you can help then please contact support@hmn.md or visit http://translate.hmn.md/.

## Installation ##

1. Install BackUpWordPress either via the WordPress.org plugin directory, or by uploading the files to your server.
2. Activate the plugin.
3. Sit back and relax safe in the knowledge that your whole site will be backed up every day.

The plugin will try to use the `mysqldump` and `zip` commands via shell if they are available, using these will greatly improve the time it takes to back up your site.

## Frequently Asked Questions ##


**Where does BackUpWordPress store the backup files?**

Backups are stored on your server in `/wp-content/backups`, you can change the directory.

**Important:** By default BackUpWordPress backs up everything in your site root as well as your database, this includes any non WordPress folders that happen to be in your site root. This does mean that your backup directory can get quite large.  

**What if I want to back up my site to another destination?**

**BackUpWordPress Pro supports Dropbox, Google Drive, Amazon S3, Rackspace, Azure, DreamObjects and FTP/SFTP. Check it out here:** [https://bwp.hmn.md](http://bwp.hmn.md/?utm_source=wordpress-org&utm_medium=plugin-page&utm_campaign=freeplugin "BackUpWordPress Homepage")  

**How do I restore my site from a backup?**

You need to download the latest backup file either by clicking download on the backups page or via `FTP`. `Unzip` the files and upload all the files to your server overwriting your site. You can then import the database using your hosts database management tool (likely `phpMyAdmin`).

See this guide for more details - [How to restore from backup](https://bwp.hmn.md/support-center/restore-backup/ "Go to support center").

**Does BackUpWordPress back up the backups directory?**

No.

**I'm not receiving my backups by email?**

Most servers have a filesize limit on email attachments, it's generally about 10mb. If your backup file is over that limit it won't be sent attached to the email, instead you should receive an email with a link to download the backup, if you aren't even receiving that then you likely have a mail issue on your server that you'll need to contact your host about.

**How many backups are stored by default?**

BackUpWordPress stores the last 10 backups by default.

**How long should a backup take?**

Unless your site is very large (many gigabytes) it should only take a few minutes to perform a back up, if your back up has been running for longer than an hour it's safe to assume that something has gone wrong, try de-activating and re-activating the plugin, if it keeps happening, contact support.

**What do I do if I get the wp-cron error message?**

The issue is that your `wp-cron.php` is not returning a `200` response when hit with a HTTP request originating from your own server, it could be several things, in most cases, it's an issue with the server / site.

There are some things you can test to confirm this is the issue.

     * Are scheduled posts working? (They use wp-cron as well ). 

     * Are you hosted on Heart Internet? (wp-cron may not be supported by Heart Internet, see below for work-around).

     * If you click manual backup does it work?

     * Try adding `define( 'ALTERNATE_WP_CRON', true );` to your `wp-config.php`, do automatic backups work?

     * Is your site private (I.E. is it behind some kind of authentication, maintenance plugin, .htaccess) if so wp-cron won't work until you remove it, if you are and you temporarily remove the authentication, do backups start working?

Report the results to our support team for further help. To do this, either enable suport from your Admin Dashboard (recommended), or email support@hmn.md

**How to get BackUpWordPress working in Heart Internet**

**The script to be entered into the Heart Internet cPanel is:** `/usr/bin/php5 /home/sites/yourdomain.com/public_html/wp-cron.php` (note the space between php5 and the location of the file). The file `wp-cron.php` `chmod` must be set to `711`.  

**My backups seem to be failing?**

If your backups are failing - it's commonly caused by lack of available resources on your server. The easiest way to establish this to exclude some [of] or your entire uploades folder, running a backup an if that succeeds. If so, we know it's probably a server issue. If not, report the results to our support team for further help. To do this, either enable suport from your Admin Dashboard (recommended), or email support@hmn.md

**Further Support & Feedback**

General support questions should be posted in the <a href="http://wordpress.org/tags/backupwordpress?forum_id=10">WordPress support forums, tagged with backupwordpress.</a>

For development issues, feature requests or anybody wishing to help out with development checkout <a href="https://github.com/humanmade/backupwordpress/">BackUpWordPress on GitHub.</a>

You can also tweet <a href="http://twitter.com/humanmadeltd">@humanmadeltd</a> or email support@hmn.md for further help/support.

## Screenshots ##

### 1. Manage multiple schedules. ###
![Manage multiple schedules.](http://s.wordpress.org/extend/plugins/backupwordpress/screenshot-1.png)

### 2. Choose your schedule, backup type, number of backups to keep and whether to recieve a notification email. ###
![Choose your schedule, backup type, number of backups to keep and whether to recieve a notification email.](http://s.wordpress.org/extend/plugins/backupwordpress/screenshot-2.png)

### 3. Easily manage exclude rules and see exactly which files are included and excluded from your backup. ###
![Easily manage exclude rules and see exactly which files are included and excluded from your backup.](http://s.wordpress.org/extend/plugins/backupwordpress/screenshot-3.png)


## Upgrade Notice ##

### 3.1.4 ###

* (upgrade-options) Bump version
* strtolower is redundant
* Upgrade routine - renames service settings to avoid backslashes.
* Use the Service name as the setting name
* Clear settings for schedule settings
* Fixes bug in displaying settings error notices
* Update version number
* Merge pull request #726 from humanmade/fix-fatal-error-missing-class-addons
* Leave bare minimum to avoid fatal error
* remove old main plugin class

### 3.1.3 ###

  * Fixes backwards compatibility for add-ons and avoids a Fatal Error. Please upgrade straight to this version before upgrading your add-ons.
  
### 3.0.4 ###

  * Fixes a few minor bugs. Immediate update is recommended.

### 3.0.2 ###

**  * Important:** we have dropped support for PHP 5.2, you will not be able to activate BackUpWordPress on a server running PHP versions older than PHP 5.3.29  

### 3.0.1 ###

  * This is a critical update. Fixes a bug in the core backup library. Please update immediately.

## Changelog ##

### 3.1.3 / 2015-02-04

* Keep deprecated classes in a deprecated.php file to avoid Fatal Error in addons during upgrade.

### 3.1.2 / 2015-02-03

* (simplify-bwp-file-timestamp) Simplify the file name timestamp
* Fix class file name
* (fixup-setup-class) Add missing transients to uninstall routine
* switch Travis notifications to Slack instead of Hipchat
* Do not delete backups on uninstall
* wrap in parenthesis
* Fix include paths after moving uninstall to own file herpderp
* Update BackUp class include
* Add all cases to switch
* Pass scanned files sizes through conform_dir()
* Fix some classes and requires
* (origin/issue-684, issue-684) Fix namespaces
* Remove unnecessary wildcard from regex
* (crazy-refactor) Remove some duplication
* Formatting
* Fix Class name
* Blank line
* Uses a DirectoryIterator to delete files
* Remove tests for now
* Fix stable tag number
* Revert to using uninstall.php
* Add tests for uninstall and deactivate
* Remove double lie break
* Remove double line break
* Remove phpdocumentor
* We still need to require some scripts...
* We dont need to check for this constant in the hook callback
* Use get_col to directly have an array of schedule option names
* Remove namespacing
* Remove debugging
* Make the uninstall, activation and deactivation hooks work
* (change-singleton-implementation) Change the singleton implementation to not use the static() function
* path argument interfers with wp-cli path argument.
* Fix some scrutinizer issues, mostly major ones
* Consistent default name for database dumps
* Major 5.3 re-factoring fun
* Minor code formatting
* else if should be elseif
* Update the WP_CLI command to fix some issues
* Add support for copying and updating an existing backups

#### 3.1.0 / 2015-01-13

* Correct $response1 variable
* (origin/issue-652, issue-652) Suppress warnings from filesystem functions
* Remove debugging function
* (origin/issue-475, issue-475) Fix a French string
* Format the intercom data better
* Load Intercom in the admin footer
* Fix some HTML issues
* Add a function that returns a given option value
* Return Unknown if no average exists or invalid
* Add a start parameter to function
* Pass in the start time as a parameter
* Make strings translatable
* Escape values
* Adds a function to track backup duration and another to display it
* Adds Average Backup Duration Info
* Improve display of arrays
* Fixes open_basedir warnings
* Replace spaces with tabs for indentation
* (wp-cron-test-improvements) Improve the reliability of the wp-cron test
* Pop in the directory sizes
* Adapt heartbeat pulse
* Update Javascript
* Update constant name
* Load the default translations so that activation error message is in user language
* Fix a few translatable strings
* Update French translations
* Re-uglify Intercom js
* Use provided callback argument to determine current screen
* remove deprecated code
* Only show notices on BWP admin page
* Fix class name
* Make function public for now
* Get an instance of BWP to load functions
* Fix conditional
* Show admin notice if fails to meet requirements
* Remove unneeded function
* Fix references
* Refactor main class after addition of the Setup class
* Introduce a setup class
* Determine if we need to run a cleanup routine
* Introduce a custom hook for addons
* Refactor main plugin file into a singleton class
* Bring activation and deactivation into main plugin class
* css changes to fix support button alignment
* Fixes issue preventing backups from running without JS
* Remove back compat memory limit define
* Refactor HMBKP_Notices to support non-persistant notices
* Minor refactor for PHP minimum requirements notice
* Clear previous backup notices when running a new backup
* Fix issues with custom paths
* Improved singleton pattern
* Display errors in network admin
* Change to text input so multiple email address are supported
* Removes HTML tag from string
* Account for new schedule
* Make upsell sentence translatable
* fix return type
* Return true if transient exists
* Check if constant is defined
* Request the site size via heartbeat send
* Pass site size through heartbeat
* Only calculate if necessary

#### 3.0.4 / 2014-12-10

  * Simplify the backup count display
  * Enqueue scripts properly
  * Change how we check directory_sizes
  * Add BackUpWordPress test case class
  * Move class to separate file
  * Group help tab display functions
  * Update POT file
  * Make strings ready for translation

#### 3.0.3 / 2014-12-06

  * Add French translations
  * Update textdomain
  * Fix deleted function
  * Use HM_Backup function to determine if shell_exec is enabled
  * Check if file is readable first
  * Fix how we kick off Task
  * Remove unused var
  * Call recursive filescanner on admin page load
  * Update package.json
  * Update the POT file task params
  * Replace hmbkp with backupwordpress as textdomain
  * Exclude folders
  * Add Grunt task options
  * Remove hard line returns
  * Display placeholder if exec fails
  * Add 30 second time diff allowance
  * Use $TRAVIS_BUILD_DIR
  * we do this in travis.yml
  * Regenerate github account info
  * Update command to reflect function rename
  * Update tests to reflect function name change
  * Rename function
  * Filter tests to run
  * Fix group flags
  * Calculate site size on page load

#### 3.0.2 / 2014-12-02

  * Bump version
  * Add parenthesis to require_once
  * Remove disk space info
  * Attempt to fix build
  * define WP_TESTS_DIR
  * Make sure current_action fnction is loaded
  * Remove extra slashes and semicolons
  * Remove the custom webhook service
  * Better default WP_TESTS_DIR
  * move trailingslash calls out of the loop for performance
  * Minor code cleanup
  * Correct path for hm-backup so it's tests are run
  * add composer.lock
  * Use scrutinizer code coverage
  * add waffle.io badge
  * Use up to date commands for coverage
  * Add code rating
  * Don't test 5.2
  * Update readme.md
  * excluded_dirs is deprecated
  * Update scrutinizer config
  * Add composer config and dev dependencies
  * Make PHPUnit generate an XML for coveralls
  * Add scrutinizer config
  * Ignore vendor dir
  * Add a link to Help page
  * Move to activation hook
  * Bump up required WP version
  * Escape all the things
  * Save errors to a notices option
  * Reload page on errors
  * Remove script
  * Remove unneeded class
  * Check PHP version on plugins loaded
  * Check PHP version on activate
  * Update readme
  * Set required version to 5.3.2
  * Update tests
  * Allow for a 30 second delta in asserting schedule time
  * Use the new build env on Travis
  * Restrict plugin to be network only
  * Fix admin URL logic
  * Formatting
  * Update readme changelog

#### 3.0.1 / 2014-11-20

  * remove uneeded images
  * latest hm-backup / fixes a serious bug in backups.
  * Typos in v 3.0 changelog
  * Specify POT filename in grunt task
  * Update POT file
  * Markdown readme
  * Regenerate readme
  * Add plugin headers

#### 3.0 / 2014-11-14

  * Refactor the recursive filesize scanner
  * correct text domain
  * Remove unneede statements
  * reload the excludes filelist in the correct place when someone excludes a file
  * Latest backdrop
  * Regenerate minified CSS
  * Regenerate minified JS
  * Fix cancel backup action
  * WordPress Coding Standards
  * Exit early if incompatible version of WordPress
  * Update translations
  * Update changelog

#### 3.0.0-beta / 2014-09-08

  * Load minified CSS
  * Formatting
  * Biweekly wording
  * Add missing period
  * use nonce_url instead of manually adding the nonce query param
  * close the settings form when done
  * Design changes as per ticket
  * Start tracking langauge in server info
  * Right align the primary button in the enable support modal
  * Re-factor the directory filesize code to use a single array instead of thoussands of transients
  * Remove the warning that would show if you were using anything other than the latest stable, it's no longer needed now that the FAQ is local
  * switch to using a single transient to store directory filesize data
  * Add an anchor link
  * Add error message
  * Rename function
  * Prefix GET params
  * Update exclude rule action
  * Adds function for building admin action urls
  * fix vertical scroll
  * Adds some functions to manage settings form submission errors
  * Rename nonce and action
  * fetch errors to display
  * Form submission handling for BWP and add-ons settings
  * Use a new function that persists form submission errors to  a transient
  * None check
  * Pass the nonce around in the ajax request
  * Enable support action links
  * Check nonces
  * Modify action URLs to use the admin_post hook
  * Remove unneeded code
  * New line at end
  * Use admin_post hook
  * Display schedule start time in local timezone
  * Add an action hook that gives access to consumers to the backup progress
 * Update translatable strings POT file
 * Bump version
 * Update min WP version and tested up to
 * accidently used dash instead of underscore
 * namespaced option and added option to uninstall.php
 * basic show/hide of premium upsell
 * Fix incorrect transient expiry
 * Show the spinner in the tab if a schedule other than the current is running
 * Snip snip
 * Bring back the some javascript enhancements
 * Switch to the PHP 5.2 compatible version of Backdrop
 * Finish up support
 * Brace up single line if's and foreach's
 * Only load minified scripts and styles if WP_DEBUG isn't on
 * Load the minified css and js files by default
 * Remove the combined css files as we only have one now. Stop loading colorbox.
 * Remove colorbox
 * Improvements
 * More improvements
 * Don't die when directly running a backup on page load
 * Re-factor the backup filesize calculation
 * make capability filterable
 * use core capabilities instead of custom
 * Switch to Backdrop
 * Fix typo in prefix
 * Hook function onto admin_post
 * Update delete link
 * Bookmark current admin page for redirect
 * Add custom capabilities and role
 * Exclude BackUpWP by default
 * Ignore the build folder
 * Add more build excludes
 * minify JS
 * generate markdown readme
 * minify css
 * combine css
 * regenerate readme
 * Make links consistent for the grunt task
 * Convert URLS to markdown
 * Update lang files
 * Add colorbox as bower dependency
 * remove colorbox folder
 * Remove colorbox submodule
 * Change how we include FAQ
 * Add readme partials
 * Add package.json
 * Add Gruntfile
 * Add bower.json
 * JSHint rules
 * remove from ignore
 * Update ignore list
 * Add bower config
 * Add FAQ strings
 * Return the cached directory size early if we have it
 * Re-factor the file browser scanner
 * correct sprintf usage
 * Handle saving service forms
 * Codeing standards
 * Move the error check outside of the foreach
 * Remove uneeded $is_tab_visible
 * Only show excludes for backups which include files
 * Show destinations in the list of schedule links
 * WordPress standard modal for enabling support
 * Remove extra slash in include path
 * Add a heading to the settings form
 * More work on excludes
 * Another todo
 * Add some todos
 * Don't include the parent directory in a browsable list
 * More work on Excludes
 * Switch to only storing 3 backups by default
 * More work on redesign
 * commented out enable support button till we figure out what to do with it
 * made tabs responsive below 639px
 * More work on redesign
 * Add more default excludes
 * Correctly highlight the tab for the first schedule
 * Switch to tabs instead of subsubsub
 * First bash at a redesign settings UX

#### 2.6.2

* Reverts a change to how the home path is calculated as it caused issues on installs where wp-config.php was stored outside of web root. Props to @mikelittle for the bug report.

#### 2.6.1

* Bump minimum WP requirement to 3.7.3, the latest security release on the 3.7 branch.
* Fix an issues that could cause schedule times to fail to account for timezone differences.
* Add a nonce check to the schedule settings.
* Fix a possible JS warning when removing an exclude rule.
* Our unit tests now run in PHP 5.2 again.

#### 2.6

* It's now possible to choose the time and day that your schedule will run on.
* Introduces several new unit tests around schedule timings.
* Fixes a bug that could cause the hourly schedule to run constantly.
* Improved the layout of the Constants help panel.
* If the backup root directory is unreadable then the plugin will no longer function.
* Update the backups table match the standard WordPress table styles.
* Improved styling for the settings dialogue.
* Improved styling for the Server Info help tab.
* /s/back ups/backups.
* Remove Deprecated call to `screen_icon`.
* Updated French translation.
* Update the `WP CLI` command to use the new method for registering command.
* Reload the schedules when re-setting up the default schedules so they show up straight away.
* s/dpesnt't/doesn't.
* Only show the estimated total schedule size when editing an existing schedule.
* Stop stripping 0 from the minutes on hourly backups so that backups at 10 (& 20, etc.) past the hour correctly show.
* Disable buttons whilst ajax requests are running.
* Move spinners outside the buttons as they didn't look very good inside.
* Improve the detection of the home path on multisite installs which have WordPress in a subdirectory.
* Track the time that the running backup is started and display how long a backup has been running for.
* Fix an issue that meant it wasn't possible to run multiple manual backups at the same time.
* Many other minor improvements.

#### 2.5

* BackUpWordPress now requires WordPress 3.7.1 as a minimum.
* Remove some old back-compat code that was required because we supported older WP versions.
* It's now possible to change the email address that notification emails are sent from using the `hmbkp_from_email` filter.
* The spinner is now retina!
* Close the PHP Session before starting the backup process to work around the 1 request per session issue. Backup status will now work on sites which happen to call `session_start`.
* Pass `max_execution_time` and the BackUpWordPress Plugin version back to support. * Include the users real name in support requests
* Stop passing `$_SERVER` with support requests as it can contain things like `.htaccess` passwords on some server configurations.
* Improve the display of the server info in the enable support popup.
* New screenshots
* Use `wp_safe_redirect` for internal redirects.
* Use `wp_is_writable` instead of `is_writable`.


#### 2.4.2

* In WordPress Multisite the backups admin page is now located in Network admin instead of the wp-admin of the main site.
* Fixed an issue with the new intercom support integration that could cause loading the backups page to timeout
* Fixed 3 stray PHP warnings.
* BackUpWordPress will now always be loaded before any BackUpWordPress Extensions.
* Fixed an issue that could cause a long modal (excludes) to show underneath the WP admin bar.

#### 2.4.1

* Add missing colorbox images

#### 2.4

* Support for new premium extensions for storing backups in a variety of online services.
* Exclude the WP DB Manager backups and WP Super Cache cache directories by default.
* We now use Intercom to offer support directly from within the plugin, opt-in of course.
* More i18n fixes / improvements.
* We no longer show download links if your backups directory isn't web accessible.
* Fix a bug that caused the plugin activation and deactivation hooks from firing.
* Correctly handle `MYSQL TIMESTAMP` columns in database dumps.
* `mysqldump` and `zip` are now correctly recognised on SmartOS.
* Schedule names are now translatable.
* Avoid having to re-calculate the filesize when a schedules type is set.
* Compatibility with WordPress 3.8


#### 2.3.2

* Correct version number.

#### 2.3.1

* Fix a PHP strict error.
* Save and close as separate buttons.
* Fix bug that caused multiple notification emails.
* Fixes typo in database option name.

changelog
da0eb54 (HEAD, origin/master, master) Merge pull request #728 from humanmade/issue-728
0417e75 Merge pull request #739 from humanmade/upgrade-options
6bcc87f (upgrade-options) Bump version
ba75dd3 strtolower is redundant
a384804 Upgrade routine
8cbe010 (origin/change-settings-names, change-settings-names) Use the Service name as the setting name
dc8b1a1 (issue-728) Cleear settings for schedule settings
54dafb4 Fixes bug in displaying settings error notices
39f1df4 (php52-compat, backup-progress) Update version number
a88d1b0 Merge pull request #726 from humanmade/fix-fatal-error-missing-class-addons
76329ce (fix-fatal-error-missing-class-addons) Leave bare minimum to avoid fatal error
4c6846d remove old main plugin class
ce4f3c8 remove deprecated notice
b37d112 Throw a deprecated warning
95fd673 (tag: 3.1.3) Merge pull request #727 from danielbachhuber/patch-1
68aea7f Fix class name
6bca5e5 Add BWP class
7b32eca Bump version
2410dcd Introduces a deprecated.php file
62d6b16 (tag: 3.1.2) Version 3.1.2
974a939 Merge pull request #687 from humanmade/fixup-setup-class
f6f6169 Merge pull request #724 from humanmade/simplify-bwp-file-timestamp
f29e319 (simplify-bwp-file-timestamp) Simplify the file name timestamp
54327e0 Fix class file name
b97f805 Merge pull request #723 from humanmade/willmot-slack-notifications
acad348 (fixup-setup-class) Add missing transients to uninstall routine
e3857e7 switch Travis notifications to Slack instead of Hipchat
2ff2b3c Do not delete backups on uninstall
2ae3630 wrap in parenthesis
0c1d81f Fix include paths after moving uninstall to own file herpderp
08ebd69 Update BackUp class include
4d25ebe Merge branch 'master' into fixup-setup-class
a79e649 (refactor-fixes, issue-688) Merge pull request #719 from humanmade/iis-fixes
d7b5b73 Merge pull request #720 from humanmade/refactor-fixes
c93c3a5 Add all cases to switch
c8a0068 Pass scanned files sizes through conform_dir()
5746c66 Fix some classes and requires
ec54f3b Merge branch 'master' into fixup-setup-class
ea32820 Merge branch 'master' into fixup-setup-class
c7e556c Merge pull request #684 from humanmade/issue-684
8ab6a83 (origin/issue-684, issue-684) Fix namespaces
9c8f826 Remove unnecessary wildcard from regex
021feb3 Merge branch 'master' into issue-684
d4c215e Merge pull request #667 from humanmade/crazy-refactor
3d8ffca (crazy-refactor) Remove some duplication
eb09901 Formatting
b020077 Fix Class name
4533bf0 Merge branch 'master' into crazy-refactor
a329b61 Blank line
2700f66 Uses a DirectoryIterator to delete files
449ef48 Merge pull request #683 from humanmade/change-singleton-implementation
219fe77 Merge branch 'master' into crazy-refactor
de70b1f Remove tests for now
d0e23a0 Fix stable tag number
fc912ee Revert to using uninstall.php
bf672f6 Add tests for uninstall and deactivate
660fd78 Remove double lie break
32c6b8e Remove double line break
f084b60 Remove phpdocumentor
51eaa6e We still need to require some scripts...
b5261a5 We dont need to check for this constant in the hook callback
867ff5d Use get_col to directly have an array of schedule option names
fee584f Remove namespacing
5130b38 Remove debugging
176c084 Make the uninstall, activation and deactivation hooks work
239f539 (change-singleton-implementation) Change the singleton implementation to not use the static() function
fd13166 (adds-filters) Merge pull request #682 from duritong/fix_path_argument
ddc75d8 path argument interfers with wp-cli path argument.
e3c2b22 (tag: v3.1.1) Version 3.1.1
b3ffe16 Merge pull request #670 from humanmade/fix-668
ad7e0fa Correct $response1 variable
89c87e8 (tag: 3.1-alpha) Merge pull request #475 from humanmade/issue-475
d0e6d0c Merge pull request #652 from humanmade/issue-652
804c472 (issue-652) Suppress warnings from filesystem functions
9861f06 Remove debugging function
6601346 Merge branch 'master' into issue-652
65808bb (issue-475) Fix a French string
ca35ba3 Format the intercom data better
c8cd859 Load Intercom in the admin footer
34c3af0 Fix some HTML issues
b896803 Fix tests
4c3fade Add a function that returns a given option value
071a23a Return Unknown if no average exists or invalid
4e6ca12 Add a start parameter to function
6d8047a Pass in the start time as a parameter
0410b01 Add some unit tests
740583a Make strings translatable
9861f70 Escape values
61a0441 Adds a function to track backup duration and anpther to display it
adb7b8a Adds Average Backup Duration Info
e9a4088 Improve display of arrays
dcbe510 Fix some scrutinizer issues, mostly major ones
cbc5a85 Consistent default name for database dumps
9e9725d Major 5.3 re-factoring fun
ef82471 Merge hm-backup into backupwordpress
4e3bfb6 Minor code formatting
6c0efd4 else if should be elseif
a970696 Update the WP_CLI command to fix some issues
35a473d Add support for copying and updating an existing backups
1ee8c69 Fixes open_basedir warnings
8de17c3 Replace spaces with tabs for indentation
082a038 Merge pull request #664 from humanmade/wp-cron-test-improvements
cc8e46e (wp-cron-test-improvements) Improve the reliability of the wp-cron test
8bf18c5 Merge pull request #639 from humanmade/issue-639
5ad724e Merge pull request #588 from humanmade/issue-588
b87396b (issue-639) Do not start schedule if backups are not possible
069553b (issue-588) Merge branch 'master' into issue-588
876b955 Pop in the directory sizes
f6d99f6 No need to return the whole array
81fc305 Merge pull request #640 from humanmade/issue-640
c11b1b5 Merge pull request #605 from humanmade/issue-605
809528e Adapt heartbeat pulse
bdb1142 Update Javascript
0f4f758 A few adjustments based on peer review feedback.
315ff75 Update constant name
195189b Load the default translations so that activation error message is in user language
cb6fbe5 Fix a few translatable strings
8c52f3a Update French translations
2f1a3a5 Re-uglify Intercom js
727579e Use provided callback argument to determine current screen
036998c remove deprecated code
34cfdc9 Merge branch 'master' into issue-605

admin/faq.php
<?php
echo '<p><strong>' . __( 'Where does BackUpWordPress store the backup files?', 'backupwordpress' ) . '</strong></p>' .

     '<p>' . __( 'Backups are stored on your server in <code>/wp-content/backups</code>, you can change the directory.', 'backupwordpress' ). '</p>' .

     '<p>' . __( 'Important: By default BackUpWordPress backs up everything in your site root as well as your database, this includes any non WordPress folders that happen to be in your site root. This does mean that your backup directory can get quite large.', 'backupwordpress' ) . '</p>' .

     '<p><strong>' . __( 'What if I want to back up my site to another destination?', 'backupwordpress' ) . '</strong></p>' .

     '<p>' . __( 'BackUpWordPress Pro supports Dropbox, Google Drive, Amazon S3, Rackspace, Azure, DreamObjects and FTP/SFTP. Check it out here: <a href="http://bwp.hmn.md/?utm_source=wordpress-org&utm_medium=plugin-page&utm_campaign=freeplugin" title="BackUpWordPress Homepage" target="_blank">https://bwp.hmn.md</a>', 'backupwordpress' ) . '</p>' .

     '<p><strong>' . __( 'How do I restore my site from a backup?', 'backupwordpress' ) . '</strong></p>' .

     '<p>' . __( 'You need to download the latest backup file either by clicking download on the backups page or via <code>FTP</code>. <code>Unzip</code> the files and upload all the files to your server overwriting your site. You can then import the database using your hosts database management tool (likely <code>phpMyAdmin</code>).', 'backupwordpress' ) . '</p>' .

     '<p>' . __( 'See this guide for more details - <a href="https://bwp.hmn.md/support-center/restore-backup/" title="Go to support center" target="_blank">How to restore from backup</a>.', 'backupwordpress' ) . '</p>' .

     '<p><strong>' . __( 'Does BackUpWordPress back up the backups directory?', 'backupwordpress' ) . '</strong></p>' .

     '<p>' . __( 'No.', 'backupwordpress' ) . '</p>' .

     '<p><strong>' . __( 'I\'m not receiving my backups by email?', 'backupwordpress' ) . '</strong></p>' .

     '<p>' . __( 'Most servers have a filesize limit on email attachments, it\'s generally about 10mb. If your backup file is over that limit it won\'t be sent attached to the email, instead you should receive an email with a link to download the backup, if you aren\'t even receiving that then you likely have a mail issue on your server that you\'ll need to contact your host about.', 'backupwordpress' ) . '</p>' .

     '<p><strong>' . __( 'How many backups are stored by default?', 'backupwordpress' ) . '</strong></p>' .

     '<p>' . __( 'BackUpWordPress stores the last 10 backups by default.', 'backupwordpress' ) . '</p>' .

     '<p><strong>' . __( 'How long should a backup take?', 'backupwordpress' ) . '</strong></p>' .

     '<p>' . __( 'Unless your site is very large (many gigabytes) it should only take a few minutes to perform a back up, if your back up has been running for longer than an hour it\'s safe to assume that something has gone wrong, try de-activating and re-activating the plugin, if it keeps happening, contact support.', 'backupwordpress' ) . '</p>' .

     '<p><strong>' . __( 'What do I do if I get the wp-cron error message?', 'backupwordpress' ) . '</strong></p>' .

     '<p>' . __( 'The issue is that your <code>wp-cron.php</code> is not returning a <code>200</code> response when hit with a HTTP request originating from your own server, it could be several things, in most cases, it\'s an issue with the server / site.', 'backupwordpress' ) . '</p>' .

     '<p>' . __( 'There are some things you can test to confirm this is the issue.', 'backupwordpress' ) . '</p>' .

     '<ul><li>' . __( 'Are scheduled posts working? (They use wp-cron as well ). ', 'backupwordpress' ) . '</li>' .

     '<li>' . __( 'Are you hosted on Heart Internet? (wp-cron may not be supported by Heart Internet, see below for work-around).', 'backupwordpress' ) . '</li>' .

     '<li>' . __( 'If you click manual backup does it work?', 'backupwordpress' ) . '</li>' .

     '<li>' . __( 'Try adding <code>define( \'ALTERNATE_WP_CRON\', true );</code> to your <code>wp-config.php</code>, do automatic backups work?', 'backupwordpress' ) . '</li>' .

     '<li>' . __( 'Is your site private (I.E. is it behind some kind of authentication, maintenance plugin, .htaccess) if so wp-cron won\'t work until you remove it, if you are and you temporarily remove the authentication, do backups start working?', 'backupwordpress' ) . '</li></ul>' .

     '<p>' . __( 'Report the results to our support team for further help. To do this, either enable suport from your Admin Dashboard (recommended), or email support@hmn.md', 'backupwordpress' ) . '</p>' .

     '<p><strong>' . __( 'How to get BackUpWordPress working in Heart Internet', 'backupwordpress' ) . '</strong></p>' .

     '<p>' . __( 'The script to be entered into the Heart Internet cPanel is: <code>/usr/bin/php5 /home/sites/yourdomain.com/public_html/wp-cron.php</code> (note the space between php5 and the location of the file). The file <code>wp-cron.php</code> <code>chmod</code> must be set to <code>711</code>.', 'backupwordpress' ) . '</p>' .

     '<p><strong>' . __( 'My backups seem to be failing?', 'backupwordpress' ) . '</strong></p>' .

     '<p>' . __( 'If your backups are failing - it\'s commonly caused by lack of available resources on your server. The easiest way to establish this to exclude some [of] or your entire uploades folder, running a backup an if that succeeds. If so, we know it\'s probably a server issue. If not, report the results to our support team for further help. To do this, either enable suport from your Admin Dashboard (recommended), or email support@hmn.md', 'backupwordpress' ) . '</p>';
